src/HealthChecks.AzureIoTHub/AzureIoTHubtHealthCheck.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Microsoft.Extensions.Diagnostics.HealthChecks;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices;
+
+namespace HealthChecks.AzureIoTHub
+{
+    public class AzureIoTHubHealthCheck : IHealthCheck
+    {
+        private readonly AzureIoTHubOptions _options;
+        public AzureIoTHubHealthCheck(AzureIoTHubOptions options)
+        {
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+        }
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                if (_options.RegistryWriteCheck)
+                {
+                    using (var client = RegistryManager.CreateFromConnectionString(_options.ConnectionString))
+                    {
+                        var deviceId = _options.RegistryWriteDeviceIdFactory();
+
+                        await client.AddDeviceAsync(new Device(deviceId), cancellationToken);
+                        await client.RemoveDeviceAsync(deviceId, cancellationToken);
+                    }
+                }
+                else if (_options.RegistryReadCheck)
```

Comment:
Are `RegistryWriteCheck` and `RegistryReadCheck` mutually exclusive options?

Reply:
Yes

you can write => you can read (always)
you can read => possibly you can write
![read](https://user-images.githubusercontent.com/4180472/68928547-5ffd9100-079b-11ea-8f21-1716181e4873.png)
![read-write](https://user-images.githubusercontent.com/4180472/68928657-ab17a400-079b-11ea-807d-1f0ceacc8467.png)